pkg/resources/resource_monitor_acceptance_test.go

```diff
@@ -1,146 +1,218 @@
+	if userEnv == "" {
+		t.Skip("Skipping TestAcc_ResourceMonitorNotifyUsers")
+	}
+	users := strings.Split(userEnv, ",")
+	name := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	config, err := resourceMonitorNotifyUsersConfig(name, users)
+	if err != nil {
+		t.Error(err)
+	}
+	checks := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "name", name),
+		resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "set_for_account", "false"),
+	}
+	for _, s := range users {
+		checks = append(checks, resource.TestCheckTypeSetElemAttr("snowflake_resource_monitor.test", "notify_users.*", s))
+	}
+	resource.ParallelTest(t, resource.TestCase{
+		Providers:    acc.TestAccProviders(),
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check:  resource.ComposeTestCheckFunc(checks...),
+			},
+			{
+				ResourceName:      "snowflake_resource_monitor.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAcc_ResourceMonitorChangeStartEndTimestamp(t *testing.T) {
+	name := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers:    acc.TestAccProviders(),
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: resourceMonitorConfigInitialTimestamp(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "name", name),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "frequency", "WEEKLY"),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "start_timestamp", "2050-01-01 12:00"),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "end_timestamp", "2055-01-01 12:00"),
+				),
+			},
+			{
+				Config: resourceMonitorConfigUpdatedTimestamp(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "name", name),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "frequency", "WEEKLY"),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "start_timestamp", "2055-01-01 12:00"),
+					resource.TestCheckResourceAttr("snowflake_resource_monitor.test", "end_timestamp", "2056-01-01 12:00"),
+				),
+			},
+			// IMPORT
+			{
+				ResourceName:      "snowflake_resource_monitor.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func resourceMonitorConfigInitialTimestamp(accName string) string {
+	return fmt.Sprintf(`
+resource "snowflake_warehouse" "warehouse" {
+  name           = "test"
+  comment        = "foo"
+  warehouse_size = "XSMALL"
+}
+
+resource "snowflake_resource_monitor" "test" {

```

Comment:
a notify_user has to be a real user (unless there's a way to mock it), so I've added a new test TestAcc_ResourceMonitorUpdateNotifyUsers, also an empty user set is ignored when properties are returned.

Reply:
I would prefer to use `ContextFunctions.CurrentUser`, because we are now not setting tests up with SNOWFLAKE_USER, but with prepared config. But we can do this cleanup after it's merged, because it's also about the other test in this file (as you already pointed out).